src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,288 @@
+    most of the main methods. Users should refer to this superclass for more information regarding those methods.
+
+    Args:
+    This class extracts rhythm and does preprocesses before being passed through the transformer model.
+        sampling_rate (`int`, *optional*, defaults to 22050):
+            Target Sampling rate of audio signal. It's the sampling rate that we forward to the model.
+        padding_value (`int`, *optional*, defaults to 0):
+            Padding value used to pad the audio. Should correspond to silences.
+        fft_window_size (`int`, *optional*, defaults to 4096):
+            Size of the window om which the Fourier transform is applied.
+        hop_length (`int`, *optional*, defaults to 1024):
+            Step between each window of the waveform.
+        frequency_min (`float`, *optional*, defaults to 10.0):
+            Minimum frequency.
+        nb_mel_filters (`int`, *optional*, defaults to 512):
+            Number of Mel filers to generate.
+        n_bars (`int`, *optional*, defaults to 2):
+            Determines interval between each sequence.
+    """
+    model_input_names = ["input_features"]
+
+    def __init__(
+        self,
+        sampling_rate: int = 22050,
+        padding_value: int = 0,
+        fft_window_size: int = 4096,
+        hop_length: int = 1024,
+        frequency_min: float = 10.0,
+        nb_mel_filters: int = 512,
+        n_bars: int = 2,
+        feature_size=None,
+        **kwargs,
+    ):
+        super().__init__(
+            feature_size=feature_size,
+            sampling_rate=sampling_rate,
+            padding_value=padding_value,
+            **kwargs,
+        )
+        self.sampling_rate = sampling_rate
+        self.padding_value = padding_value
+        self.fft_window_size = fft_window_size
+        self.hop_length = hop_length
+        self.frequency_min = frequency_min
+        self.nb_mel_filters = nb_mel_filters
+        self.n_bars = n_bars
+
+    def log_mel_spectogram(self, sequence):
+        """Generates MelSpectrogram then applies log base e."""
+
+        mel_fb = get_mel_filter_banks(
+            nb_frequency_bins=(self.fft_window_size // 2) + 1,
+            nb_mel_filters=self.nb_mel_filters,
+            frequency_min=self.frequency_min,
+            frequency_max=float(self.sampling_rate // 2),
+            sample_rate=self.sampling_rate,
+            norm=None,
+            mel_scale="htk",
+        ).astype(np.float32)
+
+        spectrogram = []
+        for seq in sequence:
+            window = np.hanning(self.fft_window_size + 1)[:-1]
+            framed_audio = fram_wave(seq, self.hop_length, self.fft_window_size)
+            spec = stft(framed_audio, window, fft_window_size=self.fft_window_size)
+            spec = np.abs(spec) ** 2.0
+            spectrogram.append(spec)
+
+        spec_shape = spec.shape
+        spectrogram = np.array(spectrogram).reshape(-1, *spec_shape)
+        log_melspec = np.log(
```

Comment:
I suggest putting the mel filterbank stuff in the constructor, since they only need to be calculated once (see Whisper's feature extractor). The rest of this function can be replaced with a call to `spectrogram()` from audio_utils.

Note: If you put the mel stuff in `__init__`, then also override `to_dict()` to prevent them from being saved to the JSON file. Again, see Whisper's feature extractor on how to do this.